False postive UWF_FIELD_NOT_INITIALIZED_IN_CONSTRUCTOR in TestNG test with @BeforeClass

SpotBugs version: 4.8.6.4 (but the bug seems to occur in all 4.8 versions)
Java: 21
Consider the attached mini project: [spotbugs.tar.gz](https://github.com/user-attachments/files/17295369/spotbugs.tar.gz)

The code passes SB versions < 4.8, but now we are getting this false positive:

```
package org.example;

import org.testng.Assert;
import org.testng.annotations.BeforeClass;
import org.testng.annotations.Test;

import java.util.concurrent.atomic.AtomicInteger;

public class SpotBugsFalsePositiveTest {

    private AtomicInteger initializedMember;

    @BeforeClass
    public void setup() {
        initializedMember = new AtomicInteger();
    }

    @Test
    public void testSyncMasterData() {
        Assert.assertEquals(initializedMember.get(), 0);
    }
}

```
Expected Behaviour: the code should pass SB, because `@BeforeClass` is supposed to be treated like a constructor. 

Actual behaviour:
```
[ERROR] Low: SpotBugsFalsePositiveTest.initializedMember not initialized in constructor and dereferenced in org.example.SpotBugsFalsePositiveTest.testSyncMasterData() [org.example.SpotBugsFalsePositiveTest] At SpotBugsFalsePositiveTest.java:[line 20] UWF_FIELD_NOT_INITIALIZED_IN_CONSTRUCTOR
```

See also #453 from 2017 where a similar false positive was then fixed.

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -25,7 +25,7 @@ Currently the versioning policy of this project follows [Semantic Versioning v2.
 - Fixed false positive `NP_NONNULL_FIELD_NOT_INITIALIZED_IN_CONSTRUCTOR` for Kotlin, handle Kotlin's `Intrinsics.checkNotNullParameter()` ([#3094](https://github.com/spotbugs/spotbugs/issues/3094))
 - Fixed some CWE mappings ([#3124](https://github.com/spotbugs/spotbugs/pull/3124))
 - Recognize some classes as immutable, fixing EI_EXPOSE and MS_EXPOSE FPs ([#3137](https://github.com/spotbugs/spotbugs/pull/3137))
-- Do not report UWF_FIELD_NOT_INITIALIZED_IN_CONSTRUCTOR for fields initialized in method annotated with TestNG's @BeforeClass. ([#3144](https://github.com/spotbugs/spotbugs/issues/3144))
+- Do not report UWF_FIELD_NOT_INITIALIZED_IN_CONSTRUCTOR for fields initialized in method annotated with TestNG's @BeforeClass. ([#3152](https://github.com/spotbugs/spotbugs/issues/3152))
 
 ### Cleanup
 - Cleanup thread issue and regex issue in test-harness ([#3130](https://github.com/spotbugs/spotbugs/issues/3130))

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -25,6 +25,7 @@ Currently the versioning policy of this project follows [Semantic Versioning v2.
 - Fixed false positive `NP_NONNULL_FIELD_NOT_INITIALIZED_IN_CONSTRUCTOR` for Kotlin, handle Kotlin's `Intrinsics.checkNotNullParameter()` ([#3094](https://github.com/spotbugs/spotbugs/issues/3094))
 - Fixed some CWE mappings ([#3124](https://github.com/spotbugs/spotbugs/pull/3124))
 - Recognize some classes as immutable, fixing EI_EXPOSE and MS_EXPOSE FPs ([#3137](https://github.com/spotbugs/spotbugs/pull/3137))
+- Do not report UWF_FIELD_NOT_INITIALIZED_IN_CONSTRUCTOR for fields initialized in method annotated with TestNG's @BeforeClass. ([#3144](https://github.com/spotbugs/spotbugs/issues/3144))
 
 ### Cleanup
 - Cleanup thread issue and regex issue in test-harness ([#3130](https://github.com/spotbugs/spotbugs/issues/3130))

--- a/spotbugs/src/main/java/edu/umd/cs/findbugs/detect/UnreadFields.java
+++ b/spotbugs/src/main/java/edu/umd/cs/findbugs/detect/UnreadFields.java
@@ -99,6 +99,7 @@ public class UnreadFields extends OpcodeStackDetector {
     private static final List<String> INITIALIZER_ANNOTATIONS = Arrays.asList(
             "Ljakarta/annotation/PostConstruct;",
             "Ljavax/annotation/PostConstruct;",
+            "Lorg/testng/annotations/BeforeClass;",
             "Lorg/junit/jupiter/api/BeforeAll;",
             "Lorg/junit/jupiter/api/BeforeEach;",
             "Lorg/junit/Before;",

